src/components/long-press/long-press.component.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Component,
+  Input,
+  Output,
+  EventEmitter,
+  ViewEncapsulation,
+  OnInit,
+  OnChanges,
+  HostBinding,
+  HostListener
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  animate,
+  transition
+} from '@angular/animations';
+
+@Component({
+  selector: 'ngx-long-press',
+  encapsulation: ViewEncapsulation.None,
+  styleUrls: ['./long-press.component.scss'],
+  host: {class: 'ngx-long-press'},
+  template: `
+    <span class="inner-background"></span>
+    <svg viewBox='-170 -170 340 340'>
+      <g transform="rotate(-90)">
+        <circle
+          r="160"
+          [@circleAnimation]="{value: pressed ? 'active' : 'inactive', params: { duration: duration }}"
+        />
+      </g>
+    </svg>
+    <button [disabled]="_disabled">
+      <ngx-icon *ngIf="active" class="icon" [fontIcon]="icon"></ngx-icon>
+      <ngx-icon *ngIf="submitted" class="icon" fontIcon="check"></ngx-icon>
+    </button>
+  `,
+  animations: [
+    trigger('circleAnimation', [
+      state('active', style({
+        strokeDasharray: '1000 1000'
+      })),
+      state('inactive', style({
+        strokeDasharray: '0 1000'
+      })),
+      transition('inactive => active', animate(`{{ duration }}ms ease-out`), {params: {duration: 1000}})
+    ])
+  ]
+})
+export class LongPressComponent implements OnInit, OnChanges {
```

Comment:
Can we break this out into a LongPressDirective (implements `(longPress)`) and the LongPressButtonComponent component?